### PR TITLE
added animation option interface to replace type any. tested various …

### PR DESCRIPTION
**Pull request by risingBirdSong**
…options out with the .to method and the options i included in the interface were the ones that were clearly animating, excluded options that seemed to have no animation effect. included offset y and x because they do animate, though they may not be appropriate in the interface. not exhaustive, may be missing options which should be included.

### CODE REVIEW

File: src/Node.ts [hunk 1]
```diff
@@ -2592,6 +2592,22 @@ export abstract class Node<Config extends NodeConfig = NodeConfig> {
   }
 }
 
+interface animOptions {
```
Comment (lavrton) [inline src/Node.ts:2595]: I don't think we should define every property. We can just extend `ShapeConfig`, `ContainerConfig`, add some tween specific properties like "duration" and ability do define any custom property.

Comment (risingBirdSong) [reply src/Node.ts:2595]: Thanks for taking a look. My motivation for the type was to get some intellisense and type checking inside the to method to help understand what valid options are.

I'll take a look at your suggestion thanks, it'll be good practice for me.

Comment (risingBirdSong) [reply src/Node.ts:2595]: Hi Anton, 

this is just an idea based on your suggestion and incomplete, but is this more appropriate?

for pick type, i only picked types i thought were relevant to animation. 

This is just idea, i am only hoping for intellisense for the user.

//pick from nodeConfig those relating to animation
type animToPick = Pick<NodeConfig, "x" | "y" | "width" | "height" | "opacity" | "scaleX" | "scaleY" | "rotation" | "rotationDeg" | "offsetX" | "offsetY">
interface nodeAndFinish {
  // the values that failed travis tests
  node: Node;
  onfinish: Function;
}

interface animToSpecific {
  //and other possible methods / properties go here 
  duration?: number;
}

type animTo = animToPick & nodeAndFinish & animToSpecific;

Thank you

Comment (lavrton) [reply src/Node.ts:2595]: Yes, something like that. But I don't think you need `Pick`, you can animate almost any property.

**Comment by lavrton** [comment]
@risingBirdSong Are you interested to finish the PR?

**Comment by risingBirdSong** [comment]
@lavrton Thanks but I don't think I can offer a high quality PR at this time.

**Comment by risingBirdSong** [comment]
@lavrton Oh I misunderstood you, I thought you meant to write a new PR implementing the better idea that we previously discussed. Thank you

**Comment by lavrton** [comment]
Yes, that what I meant. But I still merged with my own changes (from the discussion). Just to keep your contribution in history. 👍

**Comment by risingBirdSong** [comment]
Oh that makes sense, nice. Thank you 👍

**Pull request merged**